### PR TITLE
Gaurab/profiling updates

### PR DESCRIPTION
**Pull request by GaurabAryal**
### What does this PR do?

Following updates to the profiling docs for further consistency across languages and cleaned up extra details :
- Full env variable table on Java & Go tab
- Add shell example to python API key and update the env variable to `DD_API_KEY`. 
- Removed `DD_PROFILING_API_KEY`
- Removed `DD_PROFILING_TAGS` from all the languages to the default `DD_TAGS`

### Motivation

Regular cleanup and updates on the page. 

### Preview link

https://docs-staging.datadoghq.com/gaurab/profiling-updates/tracing/profiling/?tab=java
https://docs-staging.datadoghq.com/gaurab/profiling-updates/tracing/profiling/?tab=go
https://docs-staging.datadoghq.com/gaurab/profiling-updates/tracing/profiling/?tab=python

### CODE REVIEW

File: content/en/tracing/profiling/_index.md [hunk 1]
```diff
@@ -54,20 +54,26 @@ The Datadog Profiler requires [Java Flight Recorder][1]. The Datadog Profiling l
...
+| `DD_API_KEY`                                     | String        | Alternate for `-Ddd.profiling.api-key-file` argument. The [Datadog API key][4] to use when uploading profiles. |
+| `DD_PROFILING_API_KEY_FILE`                      | String        | Replacement for `-Ddd.profiling.api-key-file` argument. File that should contain the [Datadog API key][4].  |
+| `DD_SITE`                                        | String        | Destination site for your profiles (versions 0.48+). Valid options are `datadoghq.com` for Datadog US site (default), and `datadoghq.eu` for the Datadog EU site. |
+| `DD_SERVICE`                                     | String        | The Datadog [service][5] name.     |
+| `DD_ENV`                                         | String        | The Datadog [environment][6] name, for example `production`.|
+| `DD_VERSION`                                     | String        | The version of your application.                             |
+| `DD_TAGS`                                        | String        | Tags to apply to an uploaded profile. Must be a list of `<key>:<value>` separated by commas such as: `layer:api, team:intake`.  |
```
Comment (dbenamydd) [inline content/en/tracing/profiling/_index.md:68]: Ooh, I like how you got rid of the "Arguments" col!

Any reason not to include the command line version for all of these?

File: content/en/tracing/profiling/_index.md [hunk 2]
```diff
@@ -54,20 +54,26 @@ The Datadog Profiler requires [Java Flight Recorder][1]. The Datadog Profiling l
...
-| `-Ddd.profiling.api-key-file` | DD_PROFILING_API_KEY_FILE | File that should contain the API key as a string. |
-|                               | DD_PROFILING_API_KEY      | Datadog API key.                                  |
-| `-Ddd.site`                   | DD_SITE                   | Destination site for your profiles (versions 0.48+). Valid options are `datadoghq.com` for Datadog US site (default), and `datadoghq.eu` for the Datadog EU site. |
+| Environment variable                             | Type          | Description                                                                                      |
+| ------------------------------------------------ | ------------- | ------------------------------------------------------------------------------------------------ |
+| `DD_PROFILING_ENABLED`                           | Boolean       | Alternate for `-Ddd.profiling.enabled` argument. Set to `true` to enable profiling.               |
+| `DD_API_KEY`                                     | String        | Alternate for `-Ddd.profiling.api-key-file` argument. The [Datadog API key][4] to use when uploading profiles. |
```
Comment (dbenamydd) [inline content/en/tracing/profiling/_index.md:62]: IMO if we didn't support this back to the initial release, which I don't think we did, we should do what we did for python (and many projects do) and show the version when it was introduced, and keep the old setting but say it's deprecated. Otherwise someone who's running one of those early versions will follow this guide and fail.

File: content/en/tracing/profiling/_index.md [hunk 3]
```diff
@@ -110,14 +120,13 @@ The Datadog Profiler requires Python 2.7+. Memory profiling only works on Python
 
 | Environment variable                             | Type          | Description                                                                                      |
 | ------------------------------------------------ | ------------- | ------------------------------------------------------------------------------------------------ |
-| `DD_API_KEY`                                     | String        | The [Datadog API key][1] to use when uploading profiles. New in version 0.37.                                        |
+| `DD_API_KEY`                                     | String        | The [Datadog API key][1] to use when uploading profiles.                                        |
```
Comment (dbenamydd) [inline content/en/tracing/profiling/_index.md:123]: Why did you remove these? There was discussion that went into adding them, including looking at how many popular open source work, and thinking about what would best serve users of various ~personas.

File: content/en/tracing/profiling/_index.md [hunk 4]
```diff
@@ -181,11 +190,25 @@ Profiler configuration:
...
-|  WithService     | String        | The Datadog [service][4] name, for example `my-web-app`, which can be set here, or in `DD_TAGS`.             |
-|  WithEnv         | String        | The Datadog [environment][5] name, for example `production`, which can be set here, or in `DD_TAGS`.         |
-|  WithTags        | String        | The tags to apply to an uploaded profile. Must be a list of in the format `<KEY1>:<VALUE1>,<KEY2>:<VALUE2>`. |
+|  WithService     | String        | The Datadog [service][4] name, for example `my-web-app`, which can be set here, or in `DD_SERVICE`.             |
+|  WithEnv         | String        | The Datadog [environment][5] name, for example `production`, which can be set here, or in `DD_ENV`.         |
+|  WithTags        | String        | The tags to apply to an uploaded profile. Must be a list of in the format `<KEY1>:<VALUE1>,<KEY2>:<VALUE2>`. You can also set this in `DD_TAGS`. |
+|  WithSite        | String        | If your organization is on Datadog EU site, set this to `datadoghq.eu`. You can also set this in `DD_SITE`. |
```
Comment (dbenamydd) [inline content/en/tracing/profiling/_index.md:196]: Same deal w min version :-)

File: content/en/tracing/profiling/_index.md [hunk 5]
```diff
@@ -181,11 +190,25 @@ Profiler configuration:
...
+| `DD_API_KEY`                                     | String        | Alternate for `-Ddd.profiling.api-key-file` argument. The [Datadog API key][2] to use when uploading profiles. |
+| `DD_PROFILING_API_KEY_FILE`                      | String        | Replacement for `-Ddd.profiling.api-key-file` argument. File that should contain the [Datadog API key][2].  |
+| `DD_SITE`                                        | String        | If your organization is on Datadog EU site, set this to `datadoghq.eu`.                          |
+| `DD_SERVICE`                                     | String        | The Datadog [service][4] name.     |
+| `DD_ENV`                                         | String        | The Datadog [environment][5] name, for example `production`.|
+| `DD_VERSION`                                     | String        | The version of your application.                              |
+| `DD_TAGS`                                        | String        | Tags to apply to an uploaded profile. Must be a list of `<key>:<value>` separated by commas such as: `layer:api, team:intake`.  |
```
Comment (dbenamydd) [inline content/en/tracing/profiling/_index.md:210]: Is this section a mistake?

Comment (GaurabAryal) [reply content/en/tracing/profiling/_index.md:210]: Looks like we do support a few of these env variables but I just realized that I forgot to remove some 😢 . Will update!

File: content/en/tracing/profiling/_index.md (hunk 3 above)
Comment (GaurabAryal) [reply content/en/tracing/profiling/_index.md:123]: Oh sorry, I wasn't aware of that! I'll add it back :)

File: content/en/tracing/profiling/_index.md (hunk 2 above)
Comment (GaurabAryal) [reply content/en/tracing/profiling/_index.md:62]: I see! We should track those on this sheet: https://docs.google.com/spreadsheets/d/1qrz2msAPWcCD8X6zkaEHZ9ut80Uuu3x1aNexV33_cG0/edit#gid=0 

For now, would you have state on what version this is supported from? Is it 0.48 +?

File: content/en/tracing/profiling/_index.md (hunk 1 above)
Comment (GaurabAryal) [reply content/en/tracing/profiling/_index.md:68]: I added them in the notes section. Do we support the others as well as cli arguments? ie `-Ddd.tags="<key>:<val>"` etc

**Comment by GaurabAryal** [comment]
I'm going to close this PR. We should overhaul update with the agent migration next week.
